beacon/validator/src/main/java/tech/pegasys/teku/validator/coordinator/performance/DefaultPerformanceTracker.java

```diff
@@ -250,6 +250,10 @@ private SafeFuture<BlockPerformance> getBlockPerformanceForEpoch(final UInt64 cu
 
   private boolean isInHistoricBlockRoots(
       final BeaconState state, final SlotAndBlockRoot producedBlock) {
+    LOG.info(
+        "Checking if block {} is in historic block roots of the state {}",
+        producedBlock,
+        state.getBlockRoots().toString());
```

Comment:
You mentioned in the description that you left it to `debug` :) , and I'd definitely go for debug if we want to merge it.
I'd not call `.toString()` (just pass the roots) here otherwise we stringify even if the level doesn't match.
I'd also log state root (`state.hashTreeRoot()`) to check if it is the state we expect to have (which it shouldn't when the bug hits.)

this way we could ask that guy to enable the debug for the perf class.

Reply:
done, thanks for the feedback